Datadog/IntegrationUnitTests/Public/CoreTelemetryIntegrationTests.swift

```diff
@@ -204,4 +204,51 @@ class CoreTelemetryIntegrationTests: XCTestCase {
         XCTAssertNotNil(methodCalledMetric.view?.id)
         XCTAssertNotNil(methodCalledMetric.action?.id)
     }
+    
+    func testGivenRUMEnabled_effectiveSampleRateIsComposed() throws {
+        // Given
+        var config = RUM.Configuration(applicationID: .mockAny())
+        config.telemetrySampleRate = 90
+        RUM.enable(with: config, in: core)
+
+        // When
+        core.telemetry.debug("Debug Telemetry")
+        core.telemetry.error("Error Telemetry")
+        core.telemetry.metric(name: "Metric Name", attributes: [:], sampleRate: 99)
+        core.telemetry.send(telemetry: .usage(.init(event: .setUser, sampleRate: 100)))
+        core.telemetry.stopMethodCalled(
+            core.telemetry.startMethodCalled(operationName: .mockRandom(), callerClass: .mockRandom(), headSampleRate: 80),
+            tailSampleRate: 99
+        )
+
+        // Then
+        let debugEvents = core.waitAndReturnEvents(ofFeature: RUMFeature.name, ofType: TelemetryDebugEvent.self)
+        let errorEvents = core.waitAndReturnEvents(ofFeature: RUMFeature.name, ofType: TelemetryErrorEvent.self)
+        let usageEvents = core.waitAndReturnEvents(ofFeature: RUMFeature.name, ofType: TelemetryUsageEvent.self)
+        
+        if let debug = debugEvents.first,
+           debug.telemetry.message == "Debug Telemetry" {
+         
+            XCTAssertEqual(debug.effectiveSampleRate, 90)
+        }
+        
+        if let error = errorEvents.first {
+            XCTAssertEqual(error.telemetry.message, "Error Telemetry")
+            XCTAssertEqual(error.effectiveSampleRate, 90)
+        }
+
+        if let metric = debugEvents.first(where: { $0.telemetry.message == "[Mobile Metric] Metric Name" }) {
+
+            XCTAssertEqual(metric.effectiveSampleRate, 89)
+        }
+
+        if let metric = debugEvents.first(where: { $0.telemetry.message == "[Mobile Metric] Method Called" }) {
+
+            XCTAssertEqual(metric.effectiveSampleRate, 71)
```

Comment:
suggestion/ `71` looks quite magical here. What if we extract all sample rates to separate properties and define it as multiplication of certain rates?
```swift
XCTAssertEqual(metric.effectiveSampleRate, telemetrySampleRate * metricsSampleRate)
```